src/core/http/api/AuthApi.ts

```diff
@@ -0,0 +1,72 @@
+import { BaseApi } from "./BaseApi";
+import { HTTPMethod } from "./types/HTTPTypes";
+import { RequestResult } from "./types/RequestResult";
+import { SwaggerUserInfo } from "./types/SwaggerTypes";
+
+export class AuthApi extends BaseApi {
+  /**
+   * Регистрация
+   * @param data Регистрационные данные
+   */
+  signUp(data: {
+    first_name: string,
+    second_name: string,
+    login: string,
+    email: string,
+    password: string,
+    phone: string
+  }
+  ) : Promise<{id: number}> {
+    return new Promise<{id: number}>((resolve, reject) => {
+      this._request<{id: number}>('auth/signup', HTTPMethod.POST, data).then(result => {
+        resolve(result.data);
+      })
+      .catch((err: RequestResult<any>) => {
+        /** Обработку ошибок добавим позже по мере выполнения */
+        reject(err.error);
+      })
+    });
```

Comment:
this._request уже возвращает промис, его не нужно дополнительно оборачивать в `new Promise`.
Так же - если здесь нет обработки ошибок на этом уровне (Api), то блок catch здесь просто можно опустить.
```suggestion
    return this._request<{id: number}>('auth/signup', HTTPMethod.POST, data)
      .then(result => result.data);
```
Если есть обработка ошибок здесь, то ошибку можно затем прокинуть дальше
```suggestion
    return this._request<{id: number}>('auth/signup', HTTPMethod.POST, data)
      .then(result => result.data)
      .catch((err: RequestResult<any>) => {
        ...
        throw err;
      })
  });
```